crate/server/src/start_kms_server.rs

```diff
@@ -36,8 +48,168 @@ use crate::{
         ui_auth::configure_auth_routes,
     },
     socket_server::{SocketServer, SocketServerParams},
+    start_kms_server::google_cse::operations::GOOGLE_CSE_ID,
 };
 
+/// Handles the initialization or import of the Google CSE RSA keypair in the KMS.
+///
+/// This function performs the following logic:
+/// 1. Attempts to retrieve the RSA keypair for Google CSE from the KMS.
+/// 2. If the keypair exists and is valid, the function returns successfully.
+/// 3. If the keypair is not found and a migration PEM key is provided in the `ServerParams`,
+///    the key is imported along with its derived public key.
+/// 4. If no key exists and no migration key is available, a new RSA keypair is created and stored.
+///
+/// This ensures that the KMS either reuses an existing RSA keypair, imports one for migration purposes,
+/// or generates a new one, depending on availability and configuration.
+///
+/// # Arguments
+///
+/// * `kms_server` - A reference-counted pointer to the KMS instance, used to interact with the key management backend.
+/// * `server_params` - A reference-counted pointer to the server configuration, which may contain a migration key.
+///
+/// # Errors
+///
+/// Returns a `KmsError` if:
+/// * The RSA keypair fetch, import, or creation fails.
+/// * The migration PEM key is malformed or cannot be parsed.
+/// * Conversion between KMIP and OpenSSL formats fails.
+pub async fn handle_google_cse_rsa_keypair(
+    kms_server: &Arc<KMS>,
+    server_params: &Arc<ServerParams>,
+) -> KResult<()> {
+    let uid_sk = format!("{GOOGLE_CSE_ID}_rsa");
+    let uid_pk = format!("{GOOGLE_CSE_ID}_rsa_pk");
+
+    let get_request = Get {
+        unique_identifier: Some(UniqueIdentifier::TextString(uid_sk.clone())),
+        key_format_type: Some(KeyFormatType::PKCS1),
+        key_wrap_type: Some(KeyWrapType::NotWrapped),
+        key_compression_type: None,
+        key_wrapping_specification: None,
+    };
+
+    match kms_server.get(get_request, "admin", None).await {
+        Ok(resp) if matches!(resp.object_type, ObjectType::PrivateKey) => {
+            info!("RSA Keypair for Google CSE already exists.");
+            return Ok(());
+        }
+        Ok(resp) => {
+            return Err(KmsError::CryptographicError(format!(
+                "Unexpected object type for Google CSE RSA keypair: {:?}",
+                resp.object_type
+            )));
+        }
+        Err(_) => {
+            info!("RSA Keypair for Google CSE not found from existing DB.");
+        }
```

Comment:
I don't get this pattern matching: IMHO `Ok(resp)` should not be repeated.
What about this?
```
        Ok(resp) => match resp.object_type {
            ObjectType::PrivateKey => {
                info!("RSA Keypair for Google CSE already exists.");
                return Ok(());
            }
            _ => {
                return Err(KmsError::CryptographicError(format!(
                    "Unexpected object type for Google CSE RSA keypair: {:?}",
                    resp.object_type
                )));
            }
        },
        Err(_) => {
            info!("RSA Keypair for Google CSE not found from existing DB.");
        }
```